e2e: Add scenario for microshift preset

praveenkumar opened the pull request:
With this we can able to test out the microshift preset functionality.
```
$ BUNDLE_LOCATION='--bundle-location=/home/prkumar/.crc/cache/crc_microshift_libvirt_4.13.3_amd64.crcbundle' PULL_SECRET_FILE='--pull-secret-file=/home/prkumar/pull-secret' CRC_BINARY='--crc-binary=/home/prkumar/work/github/crc/out/linux-amd64/' make e2e-story-microshift

  Background:
    Given setting config property "preset" to value "microshift" succeeds                                                                                      # testsuite.go:860 -> github.com/crc-org/crc/test/e2e/testsuite.SetConfigPropertyToValueSucceedsOrFails
[sudo] password for prkumar: 
    And executing single crc setup command succeeds        

[...]
```

praveenkumar committed:
testdata: Update security context for httpd-example deployment

It should fix following error in case of microshift preset and warning
in case of openshift preset.

```
Error creating: pods "httpd-example-6ddffbfb58-r6z7d" is forbidden: violates PodSecurity "restricted:latest": seccompProfile (pod or container "httpd-24" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
```

praveenkumar committed:
Integration: Add feature file for microshift

praveenkumar committed:
Integration: Fix the usingPreexistingCluster functionality

82ca2731 refactored the code and because of that
`usingPreexistingCluster` function broken since `GodogTags` is not
assigned. This PR make sure `GodogTags` contain the provided tags.

praveenkumar committed:
Makefile: Add microshift story target

With this we can run `make e2e-story-microshift` and test only the
microshift functionaility.

gbraad commented on test/e2e/features/story_microshift.feature:
```diff
@@ -0,0 +1,35 @@
+		Then executing "curl -s http://httpd-example-testproj.apps.crc.testing" succeeds
+		And stdout should contain "Hello CRC!"
+		When executing "crc stop" succeeds
+		And starting CRC with default bundle succeeds
```
I prefer you would describe the test as:


```
Scenario: Start and expose a basic HTTP service and check after restart
```

jsliacan commented on test/e2e/features/story_microshift.feature:
```diff
@@ -0,0 +1,35 @@
+
+	# End-to-end health check
+
+	@microshift @testdata @linux @windows @mac
```
Not sure `@mac` equals to `$GOOS`. I think `@darwin` is what works.

praveenkumar committed:
Integration: Set empty string as default value for bundle and binary

The defaults doesn't make sense and also it block the correct execution
of test. If I run the integration test without these parameter then it
should use the default from the binary for bundle and $GOPATH/bin for
the binary. This PR fix it.

praveenkumar committed:
Integration: Move user provided bundle msg to right block

Currently, it is part of different block which print it out during
testing of default bundle instead user provided one.

gbraad commented on test/e2e/testsuite/testsuite.go:
```diff
@@ -31,22 +31,17 @@ var (
 	pflag.StringVar(&testWithShell, "test-shell", "", "Specifies shell to be used for the testing.")
 
-	pflag.StringVar(&bundleLocation, "bundle-location", "/path/to/bundle.crcbundle", "Path to the bundle to be used in tests")
+	pflag.StringVar(&bundleLocation, "bundle-location", "", "Path to the bundle to be used in tests")
```
👍

gbraad commented on test/e2e/testsuite/testsuite.go:
```diff
@@ -31,22 +31,17 @@ var (
+	pflag.StringVar(&bundleLocation, "bundle-location", "", "Path to the bundle to be used in tests")
 	pflag.StringVar(&pullSecretFile, "pull-secret-file", "/path/to/pull-secret", "Path to the file containing pull secret")
-	pflag.StringVar(&CRCExecutable, "crc-binary", "/path/to/binary/crc", "Path to the CRC executable to be tested")
+	pflag.StringVar(&CRCExecutable, "crc-binary", "", "Path to the CRC executable to be tested")
```
👍

praveenkumar commented:
/hold

Need to verify the actual microshift test runs or not.

praveenkumar commented:
/test e2e-microshift-crc

praveenkumar commented:
/unhold

cfergeau commented:
Unclear to me why this seems to be a wholly different test from the other ones we have? Could we share the same "httpd" test for both microshift and openshift?

praveenkumar commented:
> Unclear to me why this seems to be a wholly different test from the other ones we have? Could we share the same "httpd" test for both microshift and openshift?

@cfergeau `httpd` scenario is part of `story_openshift` feature where it expect the cluster of openshift running (as part of background). if a scenario can be abstracted which can consumed for different features then we didn't have to duplicate this test and consumed for both the scenarios but I didn't find a way.

adrianriobo commented:
I would suggest add a config flag to inform on which preset we want to run the tests (Default value to `openshift`).  

That flag can be used within the background steps (making them dynamic based on the flag). With that approach you can avoid the duplication.

cfergeau commented:
Took another look at this, and for some reason I had missed https://github.com/crc-org/crc/blob/97705a008eda801ba291fe3b9982af464d4759e0/test/e2e/features/story_openshift.feature#L13-L37 In my previous comments, I thought story_microshift.feature was totally different from the http OpenShift tests.
This is not the case, it's the same code, which answers my initial question :)

Imo it's really useful to add a comment before duplicated code to remind people when touching one of the code blocks, the other one needs to be updated. And documentingin the commit log why this was duplicated is also useful if one wants to try to improve things.
